src/admin/index.js

```diff
@@ -426,6 +429,85 @@ module.exports = ({
       .pop()
   }
 
+  /**
+   * @param {string} groupId
+   * @param {string[]} topics
+   * @param {boolean} [resolveOffsets=false]
+   * @return {Promise}
+   */
+  const fetchConsumerGroupOffsets = async ({ groupId, topics = [], resolveOffsets = false }) => {
+    if (!groupId) {
+      throw new KafkaJSNonRetriableError(`Invalid groupId ${groupId}`)
+    }
+
+    if (!Array.isArray(topics)) {
+      throw new KafkaJSNonRetriableError(`Expected topics array, got ${topics}`)
+    }
+
+    const coordinator = await cluster.findGroupCoordinator({ groupId })
+
+    let consumerOffsets
+    if (topics.length) {
+      const topicsToFetch = await Promise.all(
+        topics.map(async topic => {
+          const partitions = await findTopicPartitions(cluster, topic)
+          const partitionsToFetch = partitions.map(partition => ({ partition }))
+          return { topic, partitions: partitionsToFetch }
+        })
+      )
+
+      const { responses } = await coordinator.offsetFetch({
+        groupId,
+        topics: topicsToFetch,
+      })
+      consumerOffsets = responses
+    } else {
```

Comment:
@nirga Why do you need this if-else?
Given that you’re in this part of the code, `topics` must be array, and if its `length === 0` - `Promise.all([])` will result an empty array in `topicsToFetch`, so you’ll actually send `topics: []` to `offsetFetch` either way